internal/services/web/app_service_resource.go

```diff
@@ -379,6 +379,14 @@ func resourceAppServiceUpdate(d *pluginsdk.ResourceData, meta interface{}) error
 		return fmt.Errorf("expanding `site_config` for App Service %q (Resource Group %q): %s", id.SiteName, id.ResourceGroup, err)
 	}
 
+	// WEBSITE_VNET_ROUTE_ALL is superseded by a setting in site_config that defaults to false from 2021-02-01
+	appSettings := expandAppServiceAppSettings(d)
+	if vnetRouteAll, ok := appSettings["WEBSITE_VNET_ROUTE_ALL"]; ok {
+		if !d.HasChange("site_config.0.vnet_route_all_enabled") { // Only update the property if it's not set explicitly
+			siteConfig.VnetRouteAllEnabled = utils.Bool(strings.EqualFold(*vnetRouteAll, "true"))
+		}
+	}
```

Comment:
do we also need to remove the old setting here? if so, should we also do this in the read too?